src/FSharp.Formatting.CommandTool/ProjectCracker.fs

```diff
@@ -0,0 +1,415 @@
+        ProjectFileName : string
+        TargetPath : string option
+        IsTestProject : bool
+        IsLibrary : bool
+        IsPackable : bool
+        RepositoryUrl : string option
+        RepositoryType : string option
+        RepositoryBranch : string option
+        UsesMarkdownComments: bool
+        FsDocsCollectionNameLink : string option
+        FsDocsLicenseLink : string option
+        FsDocsLogoLink : string option
+        FsDocsLogoSource : string option
+        FsDocsNavbarPosition: string option
+        FsDocsReleaseNotesLink : string option
+        FsDocsSourceFolder : string option
+        FsDocsSourceRepository : string option
+        FsDocsTheme: string option
+        FsDocsWarnOnMissingDocs: bool
+        PackageProjectUrl : string option
+        Authors : string option
+        GenerateDocumentationFile : bool
+        //Removed because this is typically a multi-line string and dotnet-proj-info can't handle this
+        //Description : string option
+        PackageLicenseExpression : string option
+        PackageTags : string option
+        Copyright : string option
+        PackageVersion : string option
+        PackageIconUrl : string option
+        //Removed because this is typically a multi-line string and dotnet-proj-info can't handle this
+        //PackageReleaseNotes : string option
+        RepositoryCommit : string option
+    }
+
+    let crackProjectFile slnDir extraMsbuildProperties (file : string) : CrackedProjectInfo =
+
+        let projDir = Path.GetDirectoryName(file)
+        let projectAssetsJsonPath = Path.Combine(projDir, "obj", "project.assets.json")
+        if not(File.Exists(projectAssetsJsonPath)) then
+            failwithf "project '%s' not restored" file
+
+        let additionalInfo = [
+            "OutputType"
+            "IsTestProject"
+            "IsPackable"
+            "RepositoryUrl"
+            "UsesMarkdownComments"
+            "FsDocsCollectionNameLink"
+            "FsDocsLogoSource"
+            "FsDocsNavbarPosition"
+            "FsDocsTheme"
+            "FsDocsLogoLink"
+            "FsDocsLicenseLink"
+            "FsDocsReleaseNotesLink"
+            "FsDocsSourceFolder"
+            "FsDocsSourceRepository"
+            "FsDocsWarnOnMissingDocs"
+            "RepositoryType"
+            "RepositoryBranch"
+            "PackageProjectUrl"
+            "Authors"
+            "GenerateDocumentationFile"
+            //Removed because this is typically a multi-line string and dotnet-proj-info can't handle this
+            //"Description"
+            "PackageLicenseExpression"
+            "PackageTags"
+            "Copyright"
+            "PackageVersion"
+            "PackageIconUrl"
+            //Removed because this is typically a multi-line string and dotnet-proj-info can't handle this
+            //"PackageReleaseNotes"
+            "RepositoryCommit"
+        ]
+        let gp() = Inspect.getProperties ("TargetPath" :: additionalInfo)
```

Comment:
This abbreviation is too short, just use a proper name please

Reply:
It was already named like that; I left it as-is.

Reply:
ah ok :-)